crates/uv-toolchain/src/toolchain.rs

```diff
@@ -198,20 +201,24 @@ impl Toolchain {
         &self.source
     }
 
-    pub fn key(&self) -> String {
-        format!(
-            "{}-{}-{}-{}-{}",
-            self.implementation().to_string().to_ascii_lowercase(),
-            self.python_version(),
+    pub fn key(&self) -> ToolchainKey {
+        ToolchainKey::new(
+            // TODO(zanieb): We need to figure out how to handle this, I don't want to relax the types
+            // of the key to allow unknown implementations but... its nice to construct a key for
+            // system interpreters
+            ImplementationName::from_str(self.interpreter.implementation_name()).unwrap(),
```

Comment:
Would this currently panic if i had graalpython installed?

Reply:
Probably. I need to address this before merge.